numalogic/blocks/_base.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2022 The Numaproj Authors.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Generic, Union
+
+import numpy.typing as npt
+
+from numalogic.tools.types import artifact_t, state_dict_t
+
+
+class Block(Generic[artifact_t]):
```

Comment:
why not make it abc?

Reply:
I changed this and pushed. Not sure why this PR is not reflecting that

Reply:
> I changed this and pushed. Not sure why this PR is not reflecting that

looks like this is a github issue, we are also facing this problem in numaflow repo too /cc @yhl25